### ISSUE TITLE
Error when using xsys35c to unpack the 鬼畜王ランスSA.ALD

### ISSUE THREAD

**Issue by DavideSOGG** [kichikuou/xsys35c#6]
Thanks to the developers for building an unpacking tool that is friendly to translators. When using the original CD image, it can unpack SA.ALD smoothly and get an easy-to-read .ADV file. But when I take over a translation that needs to be modified (SA.ALD is not packaged by xsys35c) the program didn't work smoothly and get the following message
![capture_20230210203243843](https://user-images.githubusercontent.com/28831997/218196272-dcebcb63-76e2-4094-9580-d98202ddf7ac.jpg)
I tried to unpack it using ALD.EXE and got the same number of .SCO files as the .ADV but that's all, I can't go any further. SCO files are full of confusing characters, which is several times more difficult to translate than using ADV.
Is there any way to fix it and can SCO be further unpacked? Not surprisingly, these SCO files cannot be repackaged back into standard ALD file using xsys35c. "cannot open decompiled: Permission denied" displayed on the command.
[鬼畜王SA.zip](https://github.com/kichikuou/xsys35c/files/10711240/SA.zip)
 
[decompiled.zip](https://github.com/kichikuou/xsys35c/files/10711245/decompiled.zip)

**Comment by kichikuou** [comment]
xsys35dc only supports decompilation of CP932 Japanese encoding or UTF-8, but the `鬼畜王SA.ALD` you provided contains GB2312 encoded text (not all, some text is still in CP932).

How was this `鬼畜王SA.ALD` file created? It would probably be easier to ask the author to give you ADV source code than to try to decompile it.

**Comment by DavideSOGG** [comment]
> xsys35dc only supports decompilation of CP932 Japanese encoding or UTF-8, but the `鬼畜王SA.ALD` you provided contains GB2312 encoded text (not all, some text is still in CP932).
> 
> How was this `鬼畜王SA.ALD` file created? It would probably be easier to ask the author to give you ADV source code than to try to decompile it.

As far as I know, this file was packaged using a very old tool (Oni.exe) It looks like the original translator was modifying the Japanese characters directly in the SCO file. This is the reason why the packaged file has two character sets.
![3fdc326d55fbb2fbf5119850584a20a44623dc69.jpg](https://user-images.githubusercontent.com/28831997/218291693-1da184ea-7f7c-40d3-9fb8-e6ee33cfbbef.jpg)
The original team has long been disbanded.
If the ADV file is not available, then touching up the translation on the basis of SCO is difficult. Perhaps it would be wise that I should re-translate all the text based on the Japanese version? Of course it would take more time in comparison.

**Comment by kichikuou** [comment]
I managed to decompile the file using xsys35dc with some [dirty hacks](https://github.com/kichikuou/xsys35c/commit/aa54c3994b447eaaa5297f404f4a2c26b9962d1a).

[KR-GB2312-decompiled.zip](https://github.com/kichikuou/xsys35c/files/10715319/KR-GB2312-decompiled.zip)

Since the original data is a mixture of GB2312 and SJIS, the source files are also a mixture. For example, if you open `ﾛｰﾄﾞｾｰﾌﾞ.ADV` as GB2312, you'll see garbled characters in line 108.

![GB2312](https://user-images.githubusercontent.com/12679772/218293576-eaf9e28a-4172-4fc4-96f8-d123a9472df7.jpg)

If you open the same file as SJIS:

![SJIS](https://user-images.githubusercontent.com/12679772/218293580-6714ffcd-c7b9-4e8b-9846-604a13e4409e.jpg)


This source code cannot be recompiled with xsys35c, because xsys35c does not recognize GB2312. You'll need to convert every GB2312/SJIS string to UTF-8.

Also, aside from the character encoding issue, I'm not 100% sure that this decompiled result is correct due to the hacks I've added to the decompiler. So it may be safer to use the Japanese version as a base.
